internal/cli/cmd/create/credentials.go

```diff
@@ -0,0 +1,204 @@
+package create
+
+import (
+	"context"
+	goerrors "errors"
+
+	"connectrpc.com/connect"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/akuity/kargo/internal/cli/client"
+	"github.com/akuity/kargo/internal/cli/config"
+	"github.com/akuity/kargo/internal/cli/option"
+	"github.com/akuity/kargo/internal/credentials"
+	v1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type createCredentialsOptions struct {
+	*option.Option
+	Config config.CLIConfig
+
+	Name           string
+	Git            bool
+	Helm           bool
+	Image          bool
+	RepoURL        string
+	RepoURLPattern string
+	Username       string
+	Password       string
+}
+
+func newCreateCredentialsCommand(
+	cfg config.CLIConfig,
+	opt *option.Option,
+) *cobra.Command {
+	cmdOpts := &createCredentialsOptions{
+		Option: opt,
+		Config: cfg,
+	}
+
+	cmd := &cobra.Command{
+		Use: `credentials [--project=project] NAME \
+    (--git | --helm | --image) \
```

Comment:
`--type=<type>` was my first instinct, but I figured three separate flags that don't require values were going to work better with completions.

I don't foresee the number of credential types growing dramatically, but if it grows by even one, then there could be a benefit to this being dynamic -- otherwise one ends up in a position where they _need_ the latest client in order to create credentials of a newer type.

Right now, I feel inclined to keep `--git`, `--helm`, and `--image`, but _also_ add `--type=<type>` and use "one of" semantics for the four of those flags.

Reply:
Fixed in a051595b97541cab94bb7e8f3be72f480a7f8022